packages/manager/modules/pci/src/projects/project/storages/databases/fork/fork.controller.js

```diff
@@ -0,0 +1,260 @@
+    this.prepareOrderData();
+    this.trackDatabases('configuration_fork', 'page');
+  }
+
+  getAvailableNetworks(region) {
+    return [
+      ...sortBy(
+        map(
+          filter(this.privateNetworks, (network) =>
+            find(
+              network.regions,
+              (r) =>
+                r.region.startsWith(region.name) &&
+                r.status === 'ACTIVE' &&
+                r.openstackId === this.database.networkId,
+            ),
+          ),
+          (privateNetwork) => ({
+            ...privateNetwork,
+            name: `${privateNetwork.vlanId.toString().padStart(4, '0')} - ${
+              privateNetwork.name
+            }`,
+          }),
+        ),
+        ['name'],
+      ),
+    ];
+  }
+
+  getSubnets(region) {
+    this.loadingSubnets = true;
+    this.DatabaseService.getSubnets(
+      this.projectId,
+      this.availablePrivateNetworks[0].id,
+    )
+      .then((subnets) => {
+        this.availableSubnets = [
+          ...map(
+            filter(subnets, (subnet) =>
+              find(subnet.ipPools, (ipPool) =>
+                ipPool.region.startsWith(region.name),
+              ),
+            ),
+            (subnet) => ({
+              ...subnet,
+              name: `${subnet.ipPools[0].network} - ${subnet.ipPools[0].region}`,
+            }),
+          ),
+        ];
+        this.model.subnet = find(
+          this.availableSubnets,
+          (subnet) => subnet.id === this.database.subnetId,
+        );
+      })
+      .catch(() => {
+        this.availableSubnets = [];
+      })
+      .finally(() => {
+        this.loadingSubnets = false;
+      });
+  }
+
+  acceptLab(accepted) {
+    this.labAccepted = accepted;
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe(this.messageContainer);
+    this.messageHandler = this.CucCloudMessage.subscribe(
+      this.messageContainer,
+      { onMessage: () => this.refreshMessages() },
+    );
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  onEngineSelect() {
+    this.model.plan = find(
+      this.model.engine.selectedVersion.plans,
+      (plan) => plan.name === this.model.plan.name,
+    );
+    this.onPlanSelect();
+  }
+
+  onPlanSelect() {
+    this.model.region = find(
+      this.model.plan.regions,
+      (region) => region.name === this.model.region.name,
+    );
+    this.onRegionSelect();
+  }
+
+  onRegionSelect() {
+    this.model.flavor = find(
+      this.model.region.flavors,
+      (flavor) => flavor.name === this.model.flavor.name,
+    );
+    this.onFlavorSelect();
+  }
+
+  onFlavorSelect() {
+    this.model.usePrivateNetwork = !this.model.flavor.supportsPublicNetwork;
+    this.model.privateNetwork = this.defaultPrivateNetwork;
+    this.model.subnet = null;
+    this.model.name = `${this.model.engine.name}-${this.model.flavor.name}-${this.model.plan.name}-`;
+    this.prepareOrderData();
+  }
+
+  prepareOrderData() {
+    this.orderData = {
```

Comment:
Seems like a replica of the code from the order page. Can we try to move these into the database classes or other classes to minimize code duplication?

Reply:
The views forks and add databases are quite different, and will diverge even more with future lot, that is why they are separated files.